### PR TITLE
Add email verification

### LINKED ISSUE #109: add email to backend
Currently in our UserManager class the methods for forgot password and email confirmation after signup just print a message to the logs with the generated token:

https://github.com/ssciwr/mondey/blob/d4ea8f85283c81e378988904473b4a3d4cd2e9b7/mondey_backend/src/mondey_backend/users.py#L23-L38

We need to modify them to send an email to the user.

I've previously set up a send-only email server for websites using this docker image: https://hub.docker.com/r/boky/postfix

But maybe there is already an email server / account associated with mondey.de that we can/should use instead?

### PR DESCRIPTION
**Pull request by lkeegan**
- docker-compose
  - add postfix image
- backend
  - add SMTP_HOST to settings
  - send user an email with a verification link when they sign up
- frontend
  - add /verify route
- resolves #109

### CODE REVIEW

File: mondey_backend/src/mondey_backend/users.py [hunk 1]
```diff
@@ -18,29 +18,30 @@
...
 from .settings import app_settings
 
 
+def send_email_validation_link(email: str, token: str) -> None:
+    msg = EmailMessage()
+    msg["From"] = "no-reply@mondey.lkeegan.dev"
+    msg["To"] = email
+    msg["Subject"] = "MONDEY-Konto aktivieren"
+    msg.set_content(
+        f"Bitte klicken Sie hier, um Ihr MONDEY-Konto zu aktivieren:\n\nhttps://mondey.lkeegan.dev/verify/{token}"
```
Comment (MaHaWo) [inline mondey_backend/src/mondey_backend/users.py:32]: should this be localized in some way or contain an english translation?

File: docker-compose.yml [hunk 2]
```diff
@@ -28,9 +28,7 @@ services:
...
-#  email:
-#    image: "boky/postfix"
-#    environment:
-#      - ALLOW_EMPTY_SENDER_DOMAINS="true"
-#    networks:
-#      - mondey-network
+  email:
+    image: "boky/postfix"
+    environment:
+      - ALLOW_EMPTY_SENDER_DOMAINS="true"
```
Comment (MaHaWo) [inline docker-compose.yml:34]: I looked this up [here]( https://github.com/bokysan/docker-postfix/blob/master/README.md#allowed_sender_domains-and-allow_empty_sender_domains) ) but could only find documentation in conjunction with `ALLOWED_SENDER_DOMAINS`. Is this temporary?

File: mondey_backend/tests/routers/test_auth.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        SMTPMock.last_message = msg
+
+
+@pytest.fixture
+def smtp_mock(monkeypatch: pytest.MonkeyPatch):
+    monkeypatch.setattr(smtplib, "SMTP", SMTPMock)
+    return SMTPMock
+
+
+def test_register_new_user(public_client: TestClient, smtp_mock: SMTPMock):
```
Comment (MaHaWo) [inline mondey_backend/tests/routers/test_auth.py:30]: do we need a test for failed verification?

**Review by MaHaWo — APPROVED** [review]
lgtm, thanks for doing this. Have 3 little questions regarding code which are more for clarification. I didn´t get it to work locally though to try it out.

*Edit:* works with iwr address, but not with google?

File: docker-compose.yml (hunk 2 above)
Comment (lkeegan) [reply docker-compose.yml:34]: `ALLOWED_SENDER_DOMAINS` is I think by default empty, so setting `ALLOW_EMPTY_SENDER_DOMAINS` tells postfix to allow sending emails from any domain

File: mondey_backend/src/mondey_backend/users.py (hunk 1 above)
Comment (lkeegan) [reply mondey_backend/src/mondey_backend/users.py:32]: I guess so, I'd go with including an english translation in the email to keep this simple

**Comment by lkeegan** [comment]
> lgtm, thanks for doing this. Have 3 little questions regarding code which are more for clarification. I didn´t get it to work locally though to try it out.
> 
> _Edit:_ works with iwr address, but not with google? I also got a 404 when clicking on the link provided, but could log in anyway after that?

Yes google are quite aggressive in blocking unknown / potentially spam domains. We'll have to see if it works when sending from mondey.de or if we need an email account or maybe sign up for a mail sending service.

The link should work if you replace mondey.lkeegan.dev with localhost in the url when running locally?

File: mondey_backend/tests/routers/test_auth.py (hunk 3 above)
Comment (lkeegan) [reply mondey_backend/tests/routers/test_auth.py:30]: yes true, will add one

**Pull request merged**